nixos/modules/services/web-apps/firefly-iii.nix

```
@@ -0,0 +1,356 @@
+        "pm.max_children" = 32;
+        "pm.start_servers" = 2;
+        "pm.min_spare_servers" = 2;
+        "pm.max_spare_servers" = 4;
+        "pm.max_requests" = 500;
+      };
+      description = mdDoc ''
+        Options for the Firefly III PHP pool. See the documentation on <literal>php-fpm.conf</literal>
+        for details on configuration directives.
+      '';
+    };
+
+    settings = mkOption {
+      default = {};
+      description = mdDoc ''
+        Options for firefly-iii configuration. Refer to
+        <https://github.com/firefly-iii/firefly-iii/blob/main/.env.example> for
+        details on supported values. All <option>_FILE values supported by
+        upstream are supported here.
+      '';
+      example = literalExpression ''
+        {
+          APP_ENV = "production";
+          APP_URL = "https://example.com";
+          APP_KEY = "SomeKeyOfExactly32Chars";
+          SITE_OWNER = "mail@example.com";
+          DB_CONNECTION = "mysql";
+          DB_HOST = "db";
+          DB_PORT = 3306;
+          DB_DATABASE = "firefly";
+          DB_USERNAME = "firefly";
+          DB_PASSWORD_FILE = "/var/secrets/firefly-iii-mysql-password.txt;
+        }
+      '';
+      type = submodule {
+        freeformType = format.type;
+        options = {
+          APP_ENV = mkOption {
+            default = "local";
+            type = enum [ "production" "local" "testing" ];
+            description = mdDoc ''
+              Top level option governing Firefly's behaviour. This is best left
+              at "local".
+            '';
+          };
+          APP_URL = mkOption {
+            description = mdDoc ''
+              The root URL that you want to host Firefly III on. All URLs
+              in Firefly III will be generated using this value.
+            '';
+            example = "https://example.com";
+            default = "http://localhost";
+            type = str;
+          };
+          DB_CONNECTION = mkOption {
+            description = mdDoc ''
+              The type of database firefly-iii would use. Can be one of
+              "pgsql", "mysql", or "sqlite".
+            '';
+            example = "mysql";
+            default = "sqlite";
+            type = enum [ "sqlite" "pgsql" "mysql" ];
+          };
+        };
+      };
+    };
+  };
+
+  #
+  # Config
+  #
+  config = mkIf cfg.enable {
+
+    # PHP
+    services.phpfpm.pools.firefly-iii = {
+      inherit user;
+      inherit group;
+      phpPackage = cfg.package.phpPackage;
+      phpOptions = ''
+        log_errors = on
+      '';
+      settings = {
+        "listen.mode" = "0660";
+        "listen.owner" = user;
+        "listen.group" = group;
+        "clear_env" = "no";
+      } // cfg.poolConfig;
+    };
+
+    systemd.services.phpfpm-firefly-iii.serviceConfig = {
+      EnvironmentFile = "/run/phpfpm/firefly-iii-env.conf";
+      ExecStartPost = "${pkgs.runtimeShell} -c 'rm /run/phpfpm/firefly-iii-env.conf'";
```

```suggestion
      ExecStartPost = "${pkgs.coreutils}/bin/rm /run/phpfpm/firefly-iii-env.conf";
```
we can create scripts easily with postStart